jcl/src/java.base/share/classes/java/lang/invoke/VarHandle.java

```diff
@@ -269,13 +276,65 @@ public static AccessMode valueFromMethodName(String methodName) {
 		COMPARE_AND_EXCHANGE,
 		GET_AND_UPDATE;
 
-		MethodType accessModeType(Class<?> param1, Class<?> param2, Class<?>... params) {
-			throw OpenJDKCompileStub.OpenJDKCompileStubThrowError();
+		/**
+		 * Gets the MethodType associated with the AccessType.
+		 * 
+		 * This method gets invoked by the derived VarHandle classes through accessModeTypeUncached.
+		 * 
+		 * OpenJ9 only uses it to retrieve the receiver class, which is not available from VarForm.
+		 * 
+		 * @param receiver class of the derived VarHandle.
+		 * @param type is the field type or value type.
+		 * @param args is the list of intermediate argument classes in the derived VarHandle's
+		 * AccessMode methods.
+		 * @return the MethodType for the corresponding AccessType.
+		 */
+		MethodType accessModeType(Class<?> receiver, Class<?> type, Class<?>... args) {
+			List<Class<?>> paramList = new ArrayList<>();
+			Class<?> returnType = null;
+			switch (this) {
+			case GET:
+				returnType = type;
+				paramList.add(receiver);
+				Collections.addAll(paramList, args);
+				break;
+			case SET:
+				returnType = void.class;
+				paramList.add(receiver);
+				Collections.addAll(paramList, args);
+				paramList.add(type);
+				break;
+			case COMPARE_AND_SET:
+				returnType = boolean.class;
+				paramList.add(receiver);
+				Collections.addAll(paramList, args);
+				Collections.addAll(paramList, type, type);
+				break;
+			case COMPARE_AND_EXCHANGE:
+				returnType = type;
+				paramList.add(receiver);
+				Collections.addAll(paramList, args);
+				Collections.addAll(paramList, type, type);
+				break;
+			case GET_AND_UPDATE:
+				returnType = type;
+				paramList.add(receiver);
+				Collections.addAll(paramList, args);
+				paramList.add(type);
+				break;
+			default:
+				throw new InternalError("Invalid AccessType.");
+			}
+			return MethodType.methodType(returnType, paramList);
 		}
 	}
 	
 	static final Unsafe _unsafe = Unsafe.getUnsafe();
 	static final Lookup _lookup = Lookup.internalPrivilegedLookup;
+
+/*[IF Java14]*/
+	static final BiFunction<String, List<Integer>, ArrayIndexOutOfBoundsException> AIOOBE_SUPPLIER = null;
```

Comment:
Is this used?

Reply:
OpenJDK `VarHandle<Type>` (X-VarHandle.java.template; generated Java code) utilizes `AIOOBE_SUPPLIER`. OpenJ9 doesn't use the generated Java code from X-VarHandle.java.template since it has its own implementation. So, `AIOOBE_SUPPLIER` is left uninitialized in OpenJ9's VarHandle class.